cmd/minishift/cmd/start_preflight_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	configCmd "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/spf13/viper"
+)
+
+var dummyIsoFile, _ = filepath.Abs(filepath.Join("tmp", "foo.iso"))
+
+func TestIsoUrl(t *testing.T) {
+	var isoURLCheck = []struct {
+		in  string
+		out bool
+	}{
+		{"http://github.com/minishift/minishift-centos/minishift-centos.iso", true},
+		{"https://github.com/minishift//minishift/minishift-b2d.iso", true},
+		{strings.Replace("file://"+dummyIsoFile, "\\", "/", -1), true},
+		{"file://home/bar/baz.iso", false},
+		{"file://minishift/hoxier/foo", false},
+		{"/home/joey/chandler/iso.iso", false},
+		{"file:///home/homie/foo", false},
+		{"ram/rahim/anthony.iso", false},
+		{"gopher/minishift/b2d", false},
+		{"file://d:/gopher/minishift/b2d", false},
+		{"c:/gohome/foo.iso", false},
+		{"http:/github.com/minishift/minishift.iso", false},
+		{"file:/home/gopher/minishift-b2d.iso", false},
+	}
+
+	createDummyIsoFiles()
+
+	for _, urlTest := range isoURLCheck {
+		viper.Set(configCmd.ISOUrl.Name, urlTest.in)
+		defer viper.Reset()
```

Comment:
-1 don't call _defer_ in a loop. _defer_ makes sure the method is guaranteed to execute when the function call ends. Calling _defer_ multiple times does not have an effect, except creating confusion.